src/Layer/C3DTilesLayer.js

```diff
@@ -328,17 +304,18 @@ class C3DTilesLayer extends GeometryLayer {
                             tileContent.tileId,
                             currentBatchId,
                             [{ start, count }], // initialize with current group
-                            batchTable.getInfoById(currentBatchId),
                             {},
                             child,
                         );
                         this.tilesC3DTileFeatures.get(tileContent.tileId).set(currentBatchId, c3DTileFeature);
                     }
                 };
 
-                for (let index = 0; index < geometryAttributes.position.array.length; index += geometryAttributes.position.itemSize) {
-                    const batchIndex = index / geometryAttributes.position.itemSize;
-                    const batchId = geometryAttributes._BATCHID.array[batchIndex];
+                const positionAttribute = child.geometry.getAttribute('position');
+                const positionAttributeSize = positionAttribute.count * positionAttribute.itemSize;
+                for (let index = 0; index < positionAttributeSize; index += positionAttribute.itemSize) {
+                    const batchIndex = index / positionAttribute.itemSize;
```

Comment:
Why are you incrementing by `posititonAttribute.itemSize` only to divide the `index` by `positionAttribute.item` in the following line?

Could you not increment it by 1 and stop the iteration at `positionAttributeSize.count`?

Reply:
I kept the existing logic but you're right it could be simplified (also `count` becomes useluss in this case). However, as the `C3DTilesStyle` widget is broken (#2267) I'd rather not do this in this PR because I cannot test my modifications.

Reply:
We are missing test cases here... Could you add a notice here and fix this code when #2267 is fixed?